deployment/cloud-run/unified_ai_api.py

```
@@ -0,0 +1,2148 @@
+    """Batch process multiple audio files for transcription."""
+    start_time = time.time()
+    results = []
+
+    try:
+        # Enforce permission always; allow pytest header override for tests only
+        if (not _has_injected_permission(request, "batch_processing") and
+                "batch_processing" not in current_user.permissions):
+            raise HTTPException(
+                status_code=403,
+                detail="Permission 'batch_processing' required"
+            )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)